tests/integration/home-page/home-page.spec.ts

```diff
@@ -244,3 +101,82 @@ describe('Home page tests on desktop', () => {
     })
   })
 })
+
+const isSearchBarValueIn = (value: string | undefined, searchBarValue: string) => {
+  return value?.toLowerCase().includes(searchBarValue.toLowerCase())
+}
+
+const isSearchBarValueInTag = (tags: string[] | undefined, searchBarValue: string) => {
+  return tags?.some((tag) => (tag.toLowerCase().includes(searchBarValue.toLowerCase())))
+}
+
+const filterDevfilesOnSearchBar = (devfiles: Devfile[], searchBarValue: string): Devfile[] => {
+  if (searchBarValue === '') {
+    return devfiles
+  }
+
+  const devfilesFilteredOnSearchBar: Devfile[] = devfiles.filter((devfile: Devfile) => {
+    if (isSearchBarValueIn(devfile.displayName, searchBarValue)) {
+      return true
+    }
+
+    if (isSearchBarValueIn(devfile.description, searchBarValue)) {
+      return true
+    }
+
+    return isSearchBarValueInTag(devfile.tags, searchBarValue)
+  })
+  return devfilesFilteredOnSearchBar
+}
+
+const filterDevfilesOnTags = (devfiles: Devfile[], tagsStateWithFreq: FilterElem[]): Devfile[] => {
```

Comment:
question for you - what is the difference between this and the one used for the actual code?

And same for the other func implementations

Reply:
The filter functions in the test do not check the state of the filterElem.